python/ql/src/semmle/python/objects/ObjectAPI.qll

```diff
@@ -505,6 +505,28 @@ abstract class FunctionValue extends CallableValue {
     predicate isOverriddenMethod() {
         exists(Value f | f.overrides(this))
     }
+    
+    /** Whether `name` is a legal argument name for this function */
+    bindingset[name]
+    predicate isLegalArgumentName(string name) {
+        this.getScope().getAnArg().asName().getId() = name
+        or
+        this.getScope().getAKeywordOnlyArg().getId() = name
+        or
+        this.getScope().hasKwArg()
+    }
+    
+    /** Whether this is a "normal" method, that is, it is exists as a class attribute 
+     *  which is not wrapped and not the __new__ method. */
```

Comment:
What does the _wrapped_ part mean? I can't see anything relating to this in the implementation :confused:

Reply:
I think your two comments answer each other. The bit about lambdas is what's meant by "wrapped".

Reply:
curious. "wrapped" makes me think of decorators.